### DISCUSSION TITLE
Detect a file with a 100% black file background and delete it off. Also, detect and delete any solid colors. Ignore file if it have any transparency.

### DISCUSSION THREAD

**Discussion by md500** [ImageMagick/ImageMagick#5328] [Help] [answered]
ImageMagick version: 7.1.0-43-Q16-HDRI-x64-dll (in releases webpage) 

OS: Windows 10 Enterprise 1809 build 17763.1012 64bit

The software have anything to detect a file with an 100% black background or any solid color like each set of rgb in a specific directory? like this?

R,G,B
0,0,255
0,255,0
0,255,255
255,00
255,0,255
255,255,0
255,255,255

Basically what I'm looking for is to find and remove any images that ONLY have a solid color background, like in the example:
![BANJO TOOIE#BC96D637#0#2_all](https://user-images.githubusercontent.com/3111925/179127098-b9efc147-60ca-4253-a882-d4be94a223e7.png)

On this directory:
![FondoNegro](https://user-images.githubusercontent.com/3111925/179128895-cb6170ef-8e1d-477f-accb-c9a2016ef19e.jpg)

NOTE: The file can be any dimension ( ?x? ) and if the file has any transparency (dont matter if is a solid color), ignore it.

Does anyone have an idea how to do it with a script or if the program has to find what I'm looking for? I have >8500 files in textures, and I think it's impossible to search images with only black solid background (or any solid color) manually.

Any help will be apreciated!

**Comment by fmw42** [comment]
You do not say what version of Imagemagick nor what platoform/OS! Please always do so as syntax may vary.

Test if the file has an std of 0 (means it is a solid color).

`convert image -colorspace gray -format "%[fx:standard_deviation<=T?1:0]" info:`

Where T is some fixed threshold value such as 0.01 (1% variation).

Or check each channel separately.

**Reply by md500** [reply]
Sorry! I'm using ver 7.1.0-43-Q16-HDRI-x64-dll (in releases webpage) 

I'm using Windows 10 Enterprise 1809 build 17763.1012 64bit

**Reply by fmw42** [reply]
`magick.exe image -colorspace gray -format "%[fx:standard_deviation<=T?1:0]" info:`

set T=0.01 for 1% off constant color tolerance or 0 if perfectly uniform color.

**Reply by md500** [reply]
Thanks, it gives me an number like 1 and 0. But when I'm try to implement this comand in a loop (on script), it prints incorrectly (unexpected print). It prints:

`stardard_deviation<=0.01?1:0]`

Instead of one and zero

Should I can send the script here to check?

I want to check this flag for each loop ( Recursive )

**Comment by fmw42** [comment, marked as answer]
If you are in a .bat script, you need to double the % to %%.  

You also need to replace "image" with the name or path to your actual image file.

You then have to test for 0 or 1. If 1, it is constant color, if 0, it is not.  Or you can just return "%[fx:standard_deviation]" and do your own testing on the value.  Sorry, I am not .bat scripter. I only script in Unix bash shell script.

**Comment by snibgo** [comment]
In Windows BAT, "%" is a special character, so when it needs to be passed to IM, it needs to be doubled, as fmw42 says.

> Should I can send the script here to check?

Feel free.

**Comment by md500** [comment]
It works!! had to make %% percent in %[fx:standard_deviation] and it works.
script works correctly flawlessly!
